packages/flutter_tools/lib/src/android/android_studio.dart

```diff
@@ -405,13 +408,25 @@ class AndroidStudio implements Comparable<AndroidStudio> {
         'AndroidStudio4.2',
         '.home',
       ));
-      if (homeDot.existsSync()) {
+      final File appInfo = globals.fs.file(globals.fs.path.join(
+        globals.platform.environment['LOCALAPPDATA']!,
+        'Google',
+        'AndroidStudio4.2',
+        '.appinfo',
+      ));
+      if (homeDot.existsSync() && appInfo.existsSync()) {
```

Comment:
Are we certain that the appinfo dot file exists for all versions that the home dot file exists for? i.e. did the first 4.2 have one?

Reply:
I have tried 4.2 Canary release (which has `AndroidStudioPreview4.2` folder in LOCALAPPDATA) it doesn't appinfo dot file, but Android Studio 4.2 Beta 1  (which has `AndroidStudioPreview4.2` folder in LOCALAPPDATA as well) does have this but here in the source, we don't search for `AndroidStudioPreview4.2` folder, we only scan `AndroidStudio4.2` folder (stable Android Studio), which has this dot file too

Reply:
Then won't this fail to detect 4.2, since it needs both homedot and appinfo now?

Reply:
yes